server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -150,6 +150,7 @@ public StoredContext stashContext() {
          */
 
         ThreadContextStruct threadContextStruct = DEFAULT_CONTEXT;
+        threadContextStruct = threadContextStruct.putPersistent(context.persistentHeaders);
```

Comment:
Assign to `threadContextStruct` above directly.

Reply:
Done, this is now chained to the call above. `DEFAULT_CONTEXT.putPersistent(context.persistentHeaders);`